### PR TITLE
 Version locked items can't be moved or changed - cherrypicked

### PR DESCRIPTION
**Pull request by jonathan-s**
Cherrypicked changes explicitly to fix this issue from #51

### CODE REVIEW

File: djangocms_navigation/admin.py [hunk 1]
```diff
@@ -165,23 +176,30 @@ def change_view(
...
-                kwargs={"menu_content_id": menu_content_id},
+
+        if self._versioning_enabled:
+            menu_content = get_object_or_404(
+                self.MenuModel._base_manager, id=menu_content_id
             )
 
+            change_perm = self.has_change_permission(request, menu_content)
+            if not change_perm:
+                messages.error(request, 'You don\'t have permission to edit or it is locked')
```
Comment (Aiky30) [inline djangocms_navigation/admin.py]: Is there a reason that this message is not using: LOCK_MESSAGE?

File: tests/test_admin.py [hunk 2]
```diff
@@ -171,12 +172,66 @@ def test_get_queryset_filters_by_content_id(self):
         )
 
 
+class MenuItemAdminVersionLocked(CMSTestCase, UsefulAssertsMixin):
+
+    def setUp(self):
+        # who is author..
```
Comment (Aiky30) [inline tests/test_admin.py]: What is the following comment for? # who is author

Comment (jonathan-s) [reply tests/test_admin.py]: I think it was for the version of the factory. But taking a look at how another version factory looked like it looks the same. So I deleted this.

**Comment by jonathan-s** [comment]
@Aiky30 Sorry it took so long to get back to this. I've fixed the issues here.

File: djangocms_navigation/admin.py [hunk 3]
```diff
@@ -165,23 +176,30 @@ def change_view(
...
-                    return HttpResponseRedirect(version_list_url(menu_content))
-                # purge menu cache
-                purge_menu_cache(site_id=menu_content.menu.site_id)
-            extra_context["list_url"] = reverse(
-                "admin:djangocms_navigation_menuitem_list",
-                kwargs={"menu_content_id": menu_content_id},
+
+        if self._versioning_enabled:
+            menu_content = get_object_or_404(
+                self.MenuModel._base_manager, id=menu_content_id
```
Comment (Aiky30) [inline djangocms_navigation/admin.py]: I can't see where self.MenuModel is defined. Is this defined anywhere?

Comment (Aiky30) [reply djangocms_navigation/admin.py]: @jonathan-s I couldn't see where this was used, it's possibly left over from changes in the other PR for making the models reusable: self.MenuModel

**Pull request merged**